ctapipe/io/zfits.py

```diff
@@ -4,109 +4,96 @@
 
 This requires the protozfitsreader python library to be installed
 """
-import logging
+from numpy import ndarray
 
-import numpy as np
-import numpy.ma as ma
-from astropy import units as u
-from astropy.time import Time
+import logging
+from ctapipe.core import Map, Field
+from ctapipe.io.containers import(
+    R0CameraContainer,
+    R0Container,
+    DataContainer
+)
+from ctapipe.io.eventsource import EventSource
 
-from ctapipe.core import Container
-from .containers import R0Container
 
 logger = logging.getLogger(__name__)
 
-try:
-    import protozfitsreader
-except ImportError as err:
-    logger.fatal(
-        "the `protozfitsreader` python module is required to access MC data: {}"
-        .format(err))
-    raise err
-
-__all__ = [
-    'zfits_event_source',
-]
-
-
-def zfits_event_source(url, max_events=None):
-    """A generator that streams data from an EventIO/HESSIO MC data file
-    (e.g. a standard CTA data file.)
-
-    Parameters
-    ----------
-    url : str
-        path to file to open
-    max_events : int, optional
-        maximum number of events to read
-
-
+__all__ = ['ZFitsEventSource']
+
+
+class ZFitsEventSource(EventSource):
```

Comment:
If an instance of `ZFitsEventSource` is created, then there is no warning if protozfitsreader is not installed. It will only fail when you start reading from the EventSource.

I would suggest creating an `__init__` method with a `try...except` where it gives some extra information about if the import of protozfitsreader failed.

For example, in `HESSIOEventSource`:

```python
    def __init__(self, config=None, tool=None, **kwargs):
        super().__init__(config=config, tool=tool, **kwargs)

        try:
            import pyhessio
        except ImportError:
            msg = "The `pyhessio` python module is required to access MC data"
            self.log.error(msg)
            raise

        self.pyhessio = pyhessio
```

Reply:
I do not think these additional 9 lines (maintenance burden, noise for the reader) are worth it.
When `protozfitsreader` is not installed, the user will get a line like this:
```
ModuleNotFoundError: No module named 'protozfitsreader'
```
I agree a message like this: "The file:{path} you are trying to read is a fits.fz file. In order to read that file you need to have a package named `protozfitsreader` installed. You can find that package here:{url}"
Contains more information. But it is worth it? 
I'd like to hear a 2nd opinion here. 

please note: it costs me less time too implement these 9 lines, than writing this answer. I just assume, in the future many people (hundreds?) will read ctapipe code, thus every line we write too many, counts hundredfold.

Reply:
My main issue is that this error won't be raised when an `ZFitsEventSource` is created, but only once an attempt is made to begin reading from it. Maybe the difference isn't so large, but in my mind it seems more right - A `ZFitsEventSource` shouldn't be allowed to exist unless `protozfitsreader` is installed.

Reply:
Ah I missed the main issue. Ok, the import protozfits is now in the `__init__`. My pylinter complains about an unused import now, which is why I did deliberately not do this in the first place....

Reply:
You could circumvent this by setting a member variable to the module in __init__, and then use that member variable in the `_generator` method instead of importing again.

Reply:
Alternatively you could use importlib:

```python
import importlib
spec = importlib.util.find_spec("protozfitsreader")
found = spam_spec is not None
```